cmake/modules/extensions.cmake

```diff
@@ -493,6 +493,94 @@ function(zephyr_library_compile_definitions item)
   target_compile_definitions(${ZEPHYR_CURRENT_LIBRARY} PRIVATE ${item} ${ARGN})
 endfunction()
 
+#
+# Helper functions for specifying _POSIX_C_SOURCE and _XOPEN_SOURCE application conformance.
```

Comment:
> Application code does not use this macros to indicate it conforms to the spec, but to indicate that it wants to use functionality defined in that version of the spec, and which it otherwise cannot assume is exposed.

That is correct.

> Conformance macros (like `_POSIX_ASYNCHRONOUS_IO`) are defined in `unistd.h` by the library which provides/implements that functionality, and indicate the provided implementation is compliant.

That is also correct.

> => All uses of "application conformance" in this PR are mixing those 2 concepts.

Not in the least. You are well off the mark with that comment.

Reply:
I'm going to mark this as resolved, since there has been no response from the commenter, and otherwise because the last point is clearly misinformed.